src/charm.py

```diff
@@ -629,6 +629,19 @@ def _external_url(self) -> str:
         """Return the externally-reachable (public) address of the alertmanager api server."""
         return self.model.config.get("web_external_url") or self.ingress.url or self._internal_url
 
+    @property
+    def _alertmanager_version(self) -> str:
+        """Returns the version of alertmanager.
+
+        Returns:
+            A string equal to the agent version.
+        """
+        container = self.unit.get_container(self._container_name)
+        if not container.can_connect():
+            return None
+        version_output, _ = container.exec([self._exe_name, "--version"]).wait_output()
+        return re.search(r"version (\d*\.\d*\.\d*)", version_output).group(1)
```

Comment:
Here too mypy should have complained about a very possible None.

Reply:
Fixed.